test/integration/node-specific/client_close.test.ts

```diff
@@ -0,0 +1,372 @@
+                });
+                const dataKey = await clientEncryption.createDataKey(provider, { masterKey });
+
+                function getEncryptExtraOptions() {
+                  if (
+                    typeof process.env.CRYPT_SHARED_LIB_PATH === 'string' &&
+                    process.env.CRYPT_SHARED_LIB_PATH.length > 0
+                  ) {
+                    return { cryptSharedLibPath: process.env.CRYPT_SHARED_LIB_PATH };
+                  }
+                  return {};
+                }
+                const schemaMap = {
+                  'db.coll': {
+                    bsonType: 'object',
+                    encryptMetadata: {
+                      keyId: [dataKey]
+                    },
+                    properties: {
+                      a: {
+                        encrypt: {
+                          bsonType: 'int',
+                          algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random',
+                          keyId: [dataKey]
+                        }
+                      }
+                    }
+                  }
+                };
+                const encryptionOptions = {
+                  autoEncryption: {
+                    keyVaultNamespace: 'keyvault.datakeys',
+                    kmsProviders,
+                    extraOptions: getEncryptExtraOptions(),
+                    schemaMap,
+                    tlsOptions: { aws: { tlsCAFile: infiniteFile } }
+                  }
+                };
+
+                const encryptedClient = new MongoClient(uri, encryptionOptions);
+                await encryptedClient.connect();
+
+                const insertPromise = encryptedClient
+                  .db('db')
+                  .collection('coll')
+                  .insertOne({ a: 1 });
+
+                chai.expect(process.getActiveResourcesInfo()).to.include('FSReqPromise');
+                log({ activeResourcesBeforeClose: process.getActiveResourcesInfo() });
+
+                await keyVaultClient.close();
+                await encryptedClient.close();
+
+                chai.expect(process.getActiveResourcesInfo()).to.include('FSReqPromise');
+                log({ activeResourcesAfterClose: process.getActiveResourcesInfo() });
+
+                const err = await insertPromise.catch(e => e);
+                chai.expect(err).to.exist;
+                chai.expect(err.errmsg).to.contain('Error in KMS response');
+              }
+            );
+          });
+        });
+      });
+
+      describe('Node.js resource: Socket', () => {
+        it('no sockets remain after client.close()', metadata, async () => {});
+      });
+    });
+  });
+
+  describe('ClientEncryption', () => {
+    describe('KMS Request', () => {
+      describe('Node.js resource: TLS file read', () => {
+        describe('when KMSRequest reads an infinite TLS file read', () => {
+          it('the file read is interrupted by client.close()', async () => {
```

Comment:
Is this right?  Either owns or referecnes a MongoClient and the MongoClient never owns a ClientEncryption.  MongoClient.close() shouldn't impact the CE's state.

I do see that Neal listed the MongoClient as the owner of the CE in his document, we can discuss next week.  For now, I'd just skip this test until we decide if we need it or not.

Reply:
A `ClientEncryption` object can't be created without a `MongoClient`, so I think the thinking here is that when that `MongoClient` is closed, this `ClientEncryption` shouldn't be making any more socket requests. What do you think?